### ISSUE TITLE
Sticky supports displaying multi-level headers

### ISSUE THREAD

**Issue by shaonianruntu** [microsoft/vscode#159264]
Currently,  sticky header can only display the outermost function name surrounding the current line. 
But most functions have function nesting, that is, a sub-function may have multiple levels of peripheral functions. 
Could the sticky header fully display the current function level, that is, display all the surrounding functions surrounding it?

For C++ Example:
```
int a(){
    int b(){
        int c(){
            int d = 1;   // cursor position
        }
    }
}
```
When my cursor is in the line `int d = 1`, 
the current sticky header can only display the outmost function, as:
```
 int a()
```
But I hoping to display all peripheral functions, as:
```
int a()
int b()
int c()
```

Thanks

**Comment by aiday-mar** [comment]
Actually it should already be possible to do that. Check the setting for stickyScroll.maxLineCount and set it to a higher number than 1. If it is already higher than 1, do tell me.

**Comment by shaonianruntu** [comment]
> Actually it should already be possible to do that. Check the setting for stickyScroll.maxLineCount and set it to a higher number than 1. If it is already higher than 1 and sticky scroll shows only one line in the given context, then there is a bug.

Thanks for your reply. 
But I cannot set this option in stable version 1.70.2.
Is this a feature for beta version?

**Comment by aiday-mar** [comment]
Ah yes, my bad. This feature is visible in the Insiders version of VS Code (https://code.visualstudio.com/insiders/) and will be in the next stable version by the end of next week.

**Comment by shaonianruntu** [comment]
Also, for multi-line display I don't know if the max level is limited. I hope that while limiting the number of lines displayed in the header to keep the interface simple, a scroll bar for the header can be added to scroll and display all levels.

> Ah yes, my bad. This feature is visible in the Insiders version of VS Code (https://code.visualstudio.com/insiders/) and will be in the next stable version by the end of next week.

Thank you, I will try later.
BTW, for multi-line display I don't know if the max level is limited. I hope that while limiting the number of lines displayed in the sticky header to keep the interface simple, a scroll bar for the sticky  header can be added to scroll and display all levels.

**Comment by aiday-mar** [comment]
Yes concerning this, in the current version there is no limit on the number of sticky lines, but in the next stable version the maximum can be set to a number from 1 to 10.

**Comment by shaonianruntu** [comment]
Thank you.

**Comment by shaonianruntu** [comment]
Hi

I update the vscode to the new version 1.71.1
<img width="278" alt="CleanShot 2022-09-02 at 13 16 33@2x" src="https://user-images.githubusercontent.com/16336682/188064026-2519f161-8e3a-4fa5-bf93-6d9224c8412e.png">

And the -maxline option can be set now, which default is -maxline = 5.
<img width="1008" alt="CleanShot 2022-09-02 at 13 16 21@2x" src="https://user-images.githubusercontent.com/16336682/188064094-822e2545-afbb-496a-b8b6-0ab1b99781db.png">

But I did a simple C++ code test:
```
#include "iostream"
using namespace std;

int main()
{
    int a = 1;
    int b = 2;
    int c = 3;
    int d = 1;
    int e = 2;
    int f = 3;
    int g = 1;
    int h = 2;
    int j = 3;
    int sync(int m)
    {
        a = m;
        b = m + 1;
        c = m + 2;
        d = 1;
        e = 2;
        f = 3;
        g = 1;
        h = 2;
        j = 3;
        int enable(int n)
        {
            g = 1;
            h = 2;
            j = 3;
            g = 1;
            h = 2;
            j = 3;
        }
    }
}
```

As shown in the figure below, it can only display the innermost functions in sticky, not all the outer functions of the current row.

![CleanShot 2022-09-02 at 13 19 03@2x](https://user-images.githubusercontent.com/16336682/188064323-019b4f54-6846-493b-b3dc-25bbbd16b191.png)

I hope it suppose to display like:
```
int main()
    int sync(int m)
        int enable(int n)
```
rather than:
```
        int enable(int n)
```

Thanks

**Comment by aiday-mar** [comment]
Hi, 

Apologies for the late reply. I suspect the reason this is happening is because typically the C++ compiler does not allow defining local functions inside of functions and hence the language service provider you are using only captures the inner-most function. Please see the following Stack Overflow answer : https://stackoverflow.com/questions/4324763/can-we-have-functions-inside-functions-in-c#:~:text=You%20cannot%20define%20a%20free%20function%20inside%20another%20in%20C%2B%2B .

**Comment by jrieken** [comment]
@shaonianruntu This depends on the language service and is out of the control of sticky scroll itself. More or less it shows the same data that the outline view and breadcrumbs use. The screen capture below is with anycode-cpp which offers from pragmatic, mostly incorrect insights

<img width="851" alt="Screenshot 2022-10-06 at 16 20 52" src="https://user-images.githubusercontent.com/1794099/194338000-324ae744-c8df-462b-8fd7-064e4354311b.png">


I am going to close this as caused by extension and I encourage you to reach out to the extension that you are using. Thanks
